.teamcity/_self/lib/customBuildType/E2EBuildType.kt

```diff
@@ -0,0 +1,170 @@
+ * @param concurrentBuilds Set the number of simultaneous builds. Defaults to 0 (infinite).
+ * @param getCalypsoLiveURL String containing the commands to be run in order to obtain the live URL. Only used for Calypso E2E.
+ * @param testGroup String corresponding to an existing group defined for Jest Runner Group.
+ * @param buildParams Environment variables and other parameters to set for the build.
+ * @param buildFeatures Features to enable on top of the default feature set (perfmon, commitStatusPublisher).
+ * @param buildTriggers Rules to trigger the build. By default, no triggers are defined.
+ * @param buildDepedencies If the build configuration depends on another existing build configuration, define it here.
+ */
+open class E2EBuildType(
+	var buildId: String,
+	var buildUuid: String,
+	var buildName: String,
+	var buildDescription: String,
+	var concurrentBuilds: Int = 0,
+	var getCalypsoLiveURL: String = "",
+	var testGroup: String,
+	var buildParams: ParametrizedWithType.() -> Unit = {},
+	var buildFeatures: BuildFeatures.() -> Unit,
+	var buildTriggers: Triggers.() -> Unit = {},
+	var buildDependencies: Dependencies.() -> Unit = {},
+
+): BuildType() {
+	init {
+		val concurrentBuilds = concurrentBuilds
+		val getCalypsoLiveURL = getCalypsoLiveURL
+		val testGroup = testGroup
+		val buildParams = buildParams
+		val buildFeatures = buildFeatures
+		val buildTriggers = buildTriggers
+		val buildDependencies = buildDependencies
+		val params = params
+
+		id( buildId )
+		uuid = buildUuid
+		name = buildName
+		description = buildDescription
+		maxRunningBuilds = concurrentBuilds
+
+		artifactRules = """
+			logs.tgz => logs.tgz
+			screenshots => screenshots
+			trace => trace
+		""".trimIndent()
+
+		vcs {
+			root(Settings.WpCalypso)
+			cleanCheckout = true
+		}
+
+		params {
+			param("env.NODE_CONFIG_ENV", "test")
+			param("env.PLAYWRIGHT_BROWSERS_PATH", "0")
+			param("env.TEAMCITY_VERSION", "2021")
+			param("env.HEADLESS", "false")
+			param("env.LOCALE", "en")
+			param("env.DEBUG", "pw:api")
+			buildParams()
+		}
+
+		steps {
+			prepareEnvironment()
+
+			bashNodeScript {
+				name = "Run tests"
+				scriptContent = """
+					# Configure bash shell.
+					shopt -s globstar
+					set -x
+
+					# For Calypso E2E build configurations, the URL environment variable
+					# is computed and exported by a script.
+					$getCalypsoLiveURL
+
+					# Enter testing directory.
+					cd test/e2e
+					mkdir temp
+
+					# Decrypt config
+					openssl aes-256-cbc -md sha1 -d -in ./config/encrypted.enc -out ./config/local-test.json -k "%E2E_CONFIG_ENCRYPTION_KEY%"
+
+					# As noted above, Calypso E2E build configurations export the URL
+					# environment variable within this step - therefore the export
+					# for NODE_CONFIG variable has to be done here instead of
+					# within the Kotlin DSL.
+					export NODE_CONFIG="{\"calypsoBaseURL\":\"${'$'}{URL%/}\"}"
```

Comment:
That's a good question - I took a look at the current config, and the extraneous `%` is present. This suggests to me that it was left over from a time when the `%URL%` configuration parameter was used.

I'll remove the extraneous %.

Reply:
I tried to further simplify how `URL` param was set and it _looked like_ it was successful on first glance.

Unfortunately I had to revert the attempts I made because the successful Calypso E2E runs were due to it being run against the standard `calypsoBaseURL: https://wordpress.com` instead of the live branches.

Until we have a better way to obtain the container name, this weird handling of URL/NODE_CONFIG variable will have to stay.